cmd/registry-replacer/main.go

```diff
@@ -509,3 +552,65 @@ func pruneReplacements(config *api.ReleaseBuildConfiguration, filter asDirective
 
 	return utilerrors.NewAggregate(errs)
 }
+
+func getPromotionTargetToDockerfileMapping(ocpBuildDataDir string, majorMinor ocpbuilddata.MajorMinor) (map[string]string, error) {
+	configs, err := ocpbuilddata.LoadImageConfigs(ocpBuildDataDir, majorMinor)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read image configs from ocp-build-data: %w", err)
+	}
+	result := map[string]string{}
+	for _, config := range configs {
+		result[config.PromotesTo()] = config.Dockerfile()
+	}
+	return result, nil
+}
+
+func updateDockerfilesToMatchOCPBuildData(
+	config *api.ReleaseBuildConfiguration,
+	promotionTargetToDockerfileMapping map[string]string,
+	majorMinorVersion string,
+	ignoredRepos sets.String,
+) {
+
+	// The tool only works for the current release
```

Comment:
Technically we just need it to not be a branch that's being fast forwarded _to_ right? So like `release-4.2` would be fine

Reply:
Right. Currently the assumption is that we will only use this going forward. We should probably make this configurable but we won't need it for older releases.